server.py

```diff
@@ -63,5 +70,34 @@ def check_errors(request):
     return 'No Errors'
 
 
+def resolve_uri(uri):
+    if os.path.isfile(uri):
+        file_content = read_file(uri)
+        guess = mimetypes.guess_type(uri)[0]
+        response = response_ok(guess, file_content)
+        return response
+    elif os.path.isdir(os.path.abspath(uri)):
+        files = file_list(uri)
+        response = response_ok('text/html', files)
+        return response
+    else:
+        return response_error('404', 'Content Not Found')
+
+
+def read_file(uri):
+    file_info = io.open(uri, "r")
+    body = file_info.read()
+    file_info.close()
+    return body
+
+
+def file_list(uri):
+    file_list = ""
+    for item in os.listdir(uri):
+        file_list += "<li>{}</li>\n".format(item)
```

Comment:
It would be better to append strings to a list and then join them than it is to concatenate strings like this.